source/administrator-guide/managing-users-and-groups/authentication-mode.rst

```diff
@@ -510,6 +511,291 @@ You can configure your environment by updating the previous file or by defining
+
+#. Create a new `App Registration`
+#. use "http://localhost:8080/geonetwork/login/oauth2/code/geonetwork-oicd" as a redirect URIs
+#. On the "Certificates & Secrets" add a new secret and record it (make sure you get the secret value and NOT the object id)
+#. Make sure the groups are in the ID token - on the "Manifest" tab, edit the JSON so that  "groupMembershipClaims": "SecurityGroup" is set
+#. On the summary page, get the Application (client) ID
+#. On the summary page, choose "Endpoints" (at the top) and get the JSON text from the "OpenID Connect metadata document" Endpoints
+
+Setup users and groups;
+
+#. In Azure AD, go to groups
+#. Add new Groups - "geonetworkAdmin", "geonetworkReviewer", etc...  Record the name and the group's **Object ID**
+#. Edit a User, and choose Groups, and add them to appropriate group.
+
+
+Your environment variables will looks like this;
+
+.. code-block:: properties
+
+    GEONETWORK_SECURITY_TYPE=openidconnect
+    OPENIDCONNECT_CLIENTSECRET='...'
+    OPENIDCONNECT_CLIENTID='...'   
+    OPENIDCONNECT_SERVERMETADATA_JSON_TEXT='...big json text...' 
+    OPENIDCONNECT_IDTOKENROLELOCATION='groups'
+    OPENIDCONNECT_ROLECONVERTER='3a94275f-7d53-4205-8d78-11f39e9ffa5a=Administrator,d93c6444-feee-4b67-8c0f-15d6796370cb=Reviewer'
+
+
+NOTE: The roles are in the "roles" part of the ID Token.
+
+
+NOTE: The OPENIDCONNECT_ROLECONVERTER converts the Azure AD Group's Object ID to a Geonetwork Role.
+
+ 
+With Application Roles  
+``````````````````````
+
+Setup the Azure Application;
+
+#. Create a new Enterprise application
+#. use "http://localhost:8080/geonetwork/login/oauth2/code/geonetwork-oicd" as a redirect URIs
+#. On the "Certificates & Secrets" add a new secret and record it (make sure you get the secret value and NOT the object id)
+#. Make sure the groups are in the ID token - on the "Manifest" tab, edit the JSON so that "groupMembershipClaims": "ApplicationGroup" is set
+#. On the summary page, get the Application (client) ID
+#. On the summary page, choose "Endpoints" (at the top) and get the JSON text from the "OpenID Connect metadata document" Endpoints
+
+Setup Application Roles;
+
+#. In Application you created, go to "App Roles".
+#. Add new Groups - "Editor", "Reviewer", etc...
+
+Assign Users:
+
+#. Go to Azure AD, Enterprise Application, then the application you created
+#. Choose "Assign users and groups"
+#. Press the "Add user/group" (top)
+#. Press "None Selected" (under Users) and choose some users
+#. Press "None Selected" (Under Select a Role) and choose some roles
+#. Configure all your users with roles
+
+
+Your environment variables will looks like this;
+
+.. code-block:: properties
+
+    GEONETWORK_SECURITY_TYPE=openidconnect
+    OPENIDCONNECT_CLIENTSECRET='...'
+    OPENIDCONNECT_CLIENTID='...'   
+    OPENIDCONNECT_SERVERMETADATA_JSON_TEXT='...big json text...' 
+    OPENIDCONNECT_IDTOKENROLELOCATION='roles'
+
+NOTE: The roles are in the "roles" part of the ID Token.
```

Comment:
Better to use Sphinx annotation `.. note::`

Reply:
changed